src/main/java/com/todo/backend/Controller.java

```diff
@@ -1,54 +1,302 @@
+
+                return diff;
+            }
+        });
+    }
+    // #endregion
+
+    // #region ################################ GET
+    @RequestMapping(method = { RequestMethod.GET }, value = { "/todos" })
+    public GETResponse getAllToDos(
+            @RequestParam(defaultValue = "1") int pag,
+            @RequestParam(required = false) Boolean done,
+            @RequestParam(required = false) String text,
+            @RequestParam(required = false) Priority priority,
+            @RequestParam(required = false) Boolean sortPriority,
+            @RequestParam(required = false) Boolean sortDueDate) {
+        // TO AVOID UNNECESARY PROCESSES, WE STORE CACHE AND CHECK FOR CHANGES
+        List<ToDo> filtered = null;
+        LastMetrics lm = this.metrics.calculate(this.DB); // CHECK METRICS AGAIN
+        Boolean isSame = this.previousParams.isSameAsPrevious(done, text, priority, sortPriority, sortDueDate);
+
+        // SAME PARAMS, AND ALREADY AN ANSWER; SO TAKE BACK THE ANSWER
+        if (isSame && lastResponse != null) {
+            filtered = lastResponse;
+        } else {
+            // DIFFERENT PARAMS OR THERE'S NOT A PREVIOUS
+            filtered = new ArrayList<>(DB);
+
+            // FILTER (IF ANY)
+            if (done != null || text != null || priority != null) {
+                filtered = this.filter(done, text, priority);
+            }
+
+            // SORT BY (IF ANY)
+            if (sortDueDate != null || sortPriority != null) {
+                sortList(filtered, sortPriority, sortDueDate);
+            }
+
+            lastResponse = filtered; // SAVE THE LIST FOR NEXT
+        }
+
+        // EVEN IF WE GOT THE SAME ANSWER, WE CAN SEND A DIFFERENT PAGE TO CLIENT
+        int maxpage = (int) Math.max(Math.ceil(filtered.size() / 10f), 1);
+        int page = (int) Math.min(Math.max(pag, 1), maxpage);
+        List<ToDo> slice = filtered.subList(10 * (page - 1), Math.min(10 * page, filtered.size()));
+
+        return (new GETResponse(slice, page, maxpage, lm));
+    }
+
+    @RequestMapping(method = { RequestMethod.GET }, value = { "/todos/{id}" })
+    public ResponseEntity<?> getToDo(@PathVariable int id) {
+        if (id < 0)
+            return new ResponseEntity<>("Invalid ID", HttpStatus.BAD_REQUEST);
+
+        for (ToDo toDo : DB) {
+            if (toDo.getId() == id)
+                return new ResponseEntity<ToDo>(toDo, HttpStatus.FOUND);
+        }
+
+        return new ResponseEntity<>("Not Found", HttpStatus.NOT_FOUND);
+    }
+    // #endregion
+
+    // #region ################################ POST
+    @RequestMapping(method = { RequestMethod.POST }, value = { "/todos" })
+    public ResponseEntity<String> createToDo(@RequestBody BasicToDo toDo) {
+        if (toDo.getText().length() <= 0 || toDo.getText().length() > 120)
+            return new ResponseEntity<>("Text length must be 120 or lower", HttpStatus.BAD_REQUEST);
+
+        boolean stat = DB.add(
+                new ToDo(
```

Comment:
for these cases use Cloneable interface and implement `clone` method at `ToDo` class, this logic should go there so we call her like:
```
DB.add(toDo.clone())
```